pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -202,6 +204,8 @@ func (s *Service) createInstance(machine *actuators.MachineScope, bootstrapToken
 		return input, err
 	}
 
+	apiServerEndpoint := fmt.Sprintf("%s:%d", machine.Network().APIServerELB.DNSName, apiServerBindPort)
```

Comment:
does this need a nil check at all or are we ok?

Reply:
Good question – I had to check, but a few lines before this we're doing

```
	if s.scope.Network().APIServerELB.DNSName == "" {
		return nil, awserrors.NewFailedDependency(
			errors.New("failed to run controlplane, APIServer ELB not available"),
		)
	}
```

So if the Network were nil at this point, it would've already panic'd there (everything else is a value & not a pointer).

That said, I'm going to look a little further into whether Network might be nil anyway, since a panic seems like the wrong response if so.

Reply:
It looks like Network() is a reference into the scope itself that'll always be valid, so we're good on that front.